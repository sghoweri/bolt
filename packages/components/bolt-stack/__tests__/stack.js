import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
} from '../../../../packages-extras/testing/testing-helpers';
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../stack.schema.yml'));
const { spacing } = schema.properties;

const timeout = 120000;

describe('<bolt-stack> component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  test('basic usage', async () => {
    const results = await renderString(
      `
      {% include "@bolt-components-stack/stack.twig" with {
        content: "This is one stack. A stack spans the full width of its parent container."
      } only %}
      {% include "@bolt-components-stack/stack.twig" with {
        content: "This is another stack."
      } only %}
    `,
      {},
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('outer CSS class via Drupal Attributes', async () => {
    const results = await render('@bolt-components-stack/stack.twig', {
      content: 'This is a stack',
      attributes: {
        class: ['u-bolt-color-teal'],
      },
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  spacing.enum.forEach(async spacingChoice => {
    test(`stack spacing: ${spacingChoice}`, async () => {
      const results = await renderString(
        `
        {% include "@bolt-components-stack/stack.twig" with {
          spacing: spacingChoice,
          content: "This is one stack. A stack spans the full width of its parent container."
        } only %}
        {% include "@bolt-components-stack/stack.twig" with {
          spacing: spacingChoice,
          content: "This is another stack."
        } only %}
      `,
        {},
      );
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });
});
