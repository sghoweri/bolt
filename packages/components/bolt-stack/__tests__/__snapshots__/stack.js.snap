// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-stack> component basic usage 1`] = `
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is one stack. A stack spans the full width of its parent container.
  </ssr-keep>
</bolt-stack>
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is another stack.
  </ssr-keep>
</bolt-stack>
`;

exports[`<bolt-stack> component outer CSS class via Drupal Attributes 1`] = `
<bolt-stack class="u-bolt-color-teal">
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is a stack
  </ssr-keep>
</bolt-stack>
`;

exports[`<bolt-stack> component stack spacing: large 1`] = `
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is one stack. A stack spans the full width of its parent container.
  </ssr-keep>
</bolt-stack>
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is another stack.
  </ssr-keep>
</bolt-stack>
`;

exports[`<bolt-stack> component stack spacing: medium 1`] = `
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is one stack. A stack spans the full width of its parent container.
  </ssr-keep>
</bolt-stack>
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is another stack.
  </ssr-keep>
</bolt-stack>
`;

exports[`<bolt-stack> component stack spacing: none 1`] = `
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is one stack. A stack spans the full width of its parent container.
  </ssr-keep>
</bolt-stack>
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is another stack.
  </ssr-keep>
</bolt-stack>
`;

exports[`<bolt-stack> component stack spacing: small 1`] = `
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is one stack. A stack spans the full width of its parent container.
  </ssr-keep>
</bolt-stack>
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is another stack.
  </ssr-keep>
</bolt-stack>
`;

exports[`<bolt-stack> component stack spacing: xlarge 1`] = `
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is one stack. A stack spans the full width of its parent container.
  </ssr-keep>
</bolt-stack>
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is another stack.
  </ssr-keep>
</bolt-stack>
`;

exports[`<bolt-stack> component stack spacing: xsmall 1`] = `
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is one stack. A stack spans the full width of its parent container.
  </ssr-keep>
</bolt-stack>
<bolt-stack>
  <ssr-keep for="bolt-stack"
            class="c-bolt-stack c-bolt-example--spacing-medium"
  >
    This is another stack.
  </ssr-keep>
</bolt-stack>
`;
